README.md

```diff
@@ -1,22 +1,161 @@
 
-* SPARK MAX Firmware v1.6.2 - Adds features that are required for swerve
-* REVLib v2023.1.2 - Includes APIs for the new firmware features
+## Development Cycle
 
-## Configuration
+### Main
 
-It is possible that this project will not work for your robot right out of the box. Various things like the CAN IDs, PIDF gains, chassis configuration, etc. must be determined for your own robot!
+Main should **always** contain known, tested, and working code that has been throroughly verified by running it on the robot. Use other development branches to create new features and test them before pulling into `main` via a Pull Request (PR). `main` cannot have code pushed to it directly, meaning a PR is **mandatory**. Furthermore, to prevent merge conflicts all branches should be based on `main`.
 
-These values can be adjusted in the `Constants.h` file.
+Use the `main` branch for non-competition usage and practice.
+
+### Competition
+
+Before a competition, create a new branch based on `main` and lock it via a GitHub protection rule to prevent pushes. Only use this branch during a competition to avoid unnecessary or breaking deployments to the RIO.
+
+:warning: ***Changes should only be made to this branch if there is a blocking error that must be fixed immediately!*** :warning:
+
+## Subsystems
+
+Following the WPILib command based structure we have broken our robot up into a number of subsystems. They are listed below:
+
+| Subsystem                                                             | Purpose                                                  |
+| :-------------------------------------------------------------------- | :------------------------------------------------------- |
+| [Drive](src/main/include/subsystems/DriveSubsystem.h)                 | Drives robot                                             |
+| [Intake](src/main/include/subsystems/IntakeSubsystem.h)               | Activates the intake                                     |
+| [Scoring](src/main/include/subsystems/ScoringSubsystem.h)             | Scores Notes                                             |
+| [LED](src/main/include/subsystems/LedSubsystem.h)                     | Wrapper for the ConnectorX moduledriver, reads from State|
+| [Climb](src/main/include/subsystems/ClimbSubsystem.h)                  | Climbs on the Stage                                      |
+
+
+## CAN IDs
+
+|     Purpose/Name     | CAN ID | Motor/Driver Type |
+| :------------------: | :----: | :---------------: |
+|   Front Right Drive  |   2    |     SparkMax      |
+|   Rear Right Drive   |   4    |     SparkMax      |
+|   Rear Left Drive    |   6    |     SparkMax      |
+|   Front Left Drive   |   8    |     SparkMax      |
+|   Front Right Turn   |   1    |     SparkMax      |
+|   Rear Right Turn    |   3    |     SparkMax      |
+|   Rear Left Turn     |   5    |     SparkMax      |
+|   Front Left Turn    |   7    |     SparkMax      |
```

Comment:
Can we add the other motors now too?

Reply:
Saving that for a future branch as any IDs beyond what’s currently documented are subject to change